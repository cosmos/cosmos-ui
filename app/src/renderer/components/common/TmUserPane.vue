<template lang="pug">
tm-list-item.tm-li-user(
  v-if="user.signedIn"
  type="link"
  to="/preferences"
  @click.native="close"
  title="Preferences"
  :subtitle="user.account")
</template>

<script>
import { mapGetters } from "vuex"
import noScroll from "no-scroll"
import { TmListItem } from "@tendermint/ui"
export default {
  name: `tm-user-pane`,
  components: { TmListItem },
  computed: { ...mapGetters([`user`, `config`]) },
  methods: {
    close() {
      this.$store.commit(`setActiveMenu`, ``)
      noScroll.off()
    }
  }
}
</script>

<style lang="stylus">
@import '~variables'

.tm-li.tm-li-user
  border-top px solid var(--bc)
  height 3.5rem + px
</style>
