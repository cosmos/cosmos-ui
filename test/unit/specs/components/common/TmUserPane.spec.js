import TmUserPane from "common/TmUserPane"
import setup from "../../../helpers/vuex-setup"

describe(`TmUserPane`, () => {
  let wrapper, store, instance
  let { mount } = setup()

  beforeEach(async () => {
    instance = mount(TmUserPane)
    store = instance.store
    wrapper = instance.wrapper
    await store.dispatch(`signIn`, {
      account: `default`,
      password: `1234567890`
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`should show the active account name`, () => {
    expect(wrapper.html()).toContain(`default`)
  })

  it(`should not show the active account name if signed out`, async () => {
    await store.dispatch(`signOut`)
    expect(wrapper.html()).toBeUndefined()
  })
})
