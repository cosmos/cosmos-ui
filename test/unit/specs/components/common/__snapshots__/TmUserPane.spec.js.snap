// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmUserPane has the expected html structure 1`] = `
<router-link-stub
  class="tm-li tm-li-link tm-li-user"
  to="/preferences"
>
  <div
    class="tm-li-container"
  >
    <div
      class="tm-li-thumb"
    >
      <!---->
    </div>
     
    <div
      class="tm-li-label"
    >
      <div
        class="tm-li-title"
      >
        Preferences
      </div>
       
      <div
        class="tm-li-subtitle"
      >
        default
      </div>
    </div>
     
    <div
      class="tm-li-icon"
    >
      <i
        class="material-icons inactive"
      >
        chevron_right
      </i>
      <i
        class="material-icons active"
      >
        my_location
      </i>
    </div>
  </div>
</router-link-stub>
`;
